src/querybuilder/metricsql.ts

```diff
@@ -0,0 +1,285 @@
+      PromOperationId.IncreasePrometheus,
+      PromOperationId.IncreasePure,
+      PromOperationId.IncreasesOverTime,
+      PromOperationId.Integrate,
+      PromOperationId.Lag,
+      PromOperationId.Lifetime,
+      PromOperationId.MadOverTime,
+      PromOperationId.MedianOverTime,
+      PromOperationId.ModeOverTime,
+      PromOperationId.RangeOverTime,
+      PromOperationId.RateOverSum,
+      PromOperationId.Rollup,
+      PromOperationId.RollupCandlestick,
+      PromOperationId.RollupDelta,
+      PromOperationId.RollupDeriv,
+      PromOperationId.RollupIncrease,
+      PromOperationId.RollupRate,
+      PromOperationId.RollupScrapeInterval,
+      PromOperationId.ScrapeInterval,
+      PromOperationId.StaleSamplesOverTime,
+      PromOperationId.StdvarOverTime,
+      PromOperationId.Sum2OverTime,
+      PromOperationId.TimestampWithName,
+      PromOperationId.TfirstOverTime,
+      PromOperationId.TlastChangeOverTime,
+      PromOperationId.TlastOverTime,
+      PromOperationId.TmaxOverTime,
+      PromOperationId.TminOverTime,
+      PromOperationId.ZscoreOverTime,
+      PromOperationId.HistogramAvg,
+      PromOperationId.HistogramStddev,
+      PromOperationId.HistogramStdvar,
+      PromOperationId.PrometheusBuckets,
+    ].map(id => createAggregationOverTime(id)),
+
+    ...[
+      PromOperationId.Abs,
+      PromOperationId.Interpolate,
+      PromOperationId.KeepLastValue,
+      PromOperationId.KeepNextValue,
+      PromOperationId.RangeAvg,
+      PromOperationId.RangeFirst,
+      PromOperationId.RangeLast,
+      PromOperationId.RangeLinearRegression,
+      PromOperationId.RangeMad,
+      PromOperationId.RangeMax,
+      PromOperationId.RangeMedian,
+      PromOperationId.RangeMin,
+      PromOperationId.RangeStddev,
+      PromOperationId.RangeStdvar,
+      PromOperationId.RangeSum,
+      PromOperationId.RangeZscore,
+      PromOperationId.RemoveResets,
+      PromOperationId.RunningAvg,
+      PromOperationId.RunningMax,
+      PromOperationId.RunningMin,
+      PromOperationId.RunningSum,
+      PromOperationId.Distinct,
+      PromOperationId.Geomean,
+      PromOperationId.Histogram
+    ].map(id => createFunction({ id })),
+
+    ...[
+      PromOperationId.BottomkAvg,
+      PromOperationId.BottomkLast,
+      PromOperationId.BottomkMax,
+      PromOperationId.BottomkMedian,
+      PromOperationId.BottomkMin,
+      PromOperationId.TopkAvg,
+      PromOperationId.TopkLast,
+      PromOperationId.TopkMax,
+      PromOperationId.TopkMedian,
+      PromOperationId.TopkMin
+    ].map(id => createFunction({
+      id: id,
+      params: [
+        { name: 'K-value', type: 'number' },
+        { name: 'other_label=other_value', type: 'string', optional: true },
+      ],
+      defaultParams: [5, ''],
+      renderer: (model: QueryBuilderOperation, def: QueryBuilderOperationDef, innerExpr: string) => {
+        return `${model.id}(${model.params[0]}, ${innerExpr}, "${model.params[1]}")`;
+      },
+    })),
```

Comment:
The function can be defined separately. 
There are a lot of different modifications made. It could be better to provide another file with those modification functions and use them in the place where it is needed.

Reply:
updated it, I changed the structure, now the creation of functions is divided by folder by type